File: 1777-determine-if-two-strings-are-close/1777-determine-if-two-strings-are-close.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        HashMap<Character,Integer> map1=new HashMap<>();
        HashMap<Character,Integer> map2=new HashMap<>();
        if(word1.length()!=word2.length())
        {
            return false;
        }

        for(char  c:word1.toCharArray())
        {
            map1.put(c,map1.getOrDefault(c,0)+1);
        }

        for(char  c:word2.toCharArray())
        {   
            if(!map1.containsKey(c))
            {
                return false;
            }
            map2.put(c,map2.getOrDefault(c,0)+1);
        }

        HashMap<Integer,Integer> hm1=new HashMap<>();
        HashMap<Integer,Integer> hm2=new HashMap<>();

        for(int freq:map1.values())
        {
            hm1.put(freq,hm1.getOrDefault(freq,0)+1);
        }
        for(int freq:map2.values())
        {
            hm2.put(freq,hm2.getOrDefault(freq,0)+1);
        }

        return hm1.equals(hm2);

        
    }
}

File: 1777-determine-if-two-strings-are-close/1777-determine-if-two-strings-are-close.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) > len(word2) or len(word1) < len(word2):
            return False
        
        word1_freq = {}
        word2_freq = {}

        for i in range(len(word1)):
            word1_freq[word1[i]] = word1_freq.get(word1[i],0)+1
            word2_freq[word2[i]] = word2_freq.get(word2[i],0)+1

        for key,value in word1_freq.items():
            if key not in word2_freq:
                return False


        word1_list = sorted(word1_freq.values())
        word2_list = sorted(word2_freq.values())

        return word1_list == word2_list




